package-lock.json

```diff
@@ -12997,7 +12997,7 @@
         "debug": "^4.1.0",
         "gensync": "^1.0.0-beta.2",
         "json5": "^2.2.2",
-        "semver": "7.5.4"
```

Comment:
I'm confused as to why package-lock.json has changed when there's no change to package.json?

Are these changes an intentional change (if so please explain why they're here in the PR description) or not (in which case please eliminate them from the PR)?

Reply:
@tackley - sorry, i missed that.

@hadiamjad - you always have to look at these things to ensure that accidental updates don't get included. in particular, you need to get `package.json` and `package-lock.json` from `main`

Reply:
This still need to be resolved - we'll end up reverting to an old version of `semver` and perhaps others when this PR is applied. Unless intentional this should be avoided.

I'm a little confused as to how this PR has 91 commits in it, dating back a very long time. It looks like work started on from a branch other than `main`?  

To resolve, I'd suggest you ensure this branch only includes commits necessary to deliver this feature. Once you've got into this state, usually an [interactive rebase](https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d) is the way to remove all the irrelevant commits: on this branch, run `git fetch` followed by `git rebase --interactive origin/main`, then in the editor window that appears delete every commit that was not part of this work. That removes them from this branch's history. 

Finally, after checking it all builds, runs, and passes tests locally as usual, do a `git push --force-with-lease` to update this branch.